freshdesk/provider/client.py

```diff
@@ -0,0 +1,50 @@
+import requests
+from flask import current_app as app
+
+from . import UpstreamProviderError
+
+
+class FreshdeskClient:
+    DEFAULT_TICKET_PARAMETERS = ["tag"]
+
+    def __init__(self, api_key, domain, parameters=None):
+        self.base_url = f"https://{domain}/api/v2"
+        # Freshdesk uses Basic Auth with this specific format, using the API key
+        self.basic_auth = (api_key, "X")
+
+        if not parameters or parameters == "":
+            parameters = self.DEFAULT_TICKET_PARAMETERS
+
+        self.parameters = parameters
+
+    def build_ticket_query(self, query):
+        """
+        Future-proofs ticket queries to allow for multiple field searches, including custom ones.
+        Joins all search fields from `TICKET_PARAMETERS` with OR conditions.
+        """
+        queries = [f"{param}:{query}" for param in self.parameters]
+        joined = " OR ".join(queries)
+
+        return f'"{joined}"'
+
+    def search(self, query):
+        search_url = f"{self.base_url}/search/tickets"
+        params = {"query": self.build_ticket_query(query)}
+
+        response = requests.get(search_url, auth=self.basic_auth, params=params)
+
+        if response.status_code != 200:
+            message = response.text or f"Error: HTTP {response.status_code}"
+            raise UpstreamProviderError(message)
+
+        return response.json()["results"]
+
+
+def get_client():
```

Comment:
Is it possible to implement the `global client` pattern in this connector? Freshdesk API doesn't support OAuth.

Reply:
Done